packages/path-provider/lib/path_provider.dart

```diff
@@ -0,0 +1,45 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:io';
+
+import 'package:flutter/services.dart';
+
+/// Returns commonly used locations on the filesystem.
+  
+const _channel = const MethodChannel('plugins.flutter.io/path_provider');
+
+  /// Path to the temporary directory on the device.
+  ///
+  /// Files in this directory may be cleared at any time. This does *not* return
+  /// a new temporary directory. Instead, the caller is responsible for creating
+  /// (and cleaning up) files or directories within this directory. This
+  /// directory is scoped to the calling application.
+  ///
+  /// On iOS, this uses the `NSTemporaryDirectory` API.
+  ///
+  /// On Android, this uses the `getCacheDir` API on the context.
+  Future<Directory> getTemporaryDirectory() async {
+    final String path = await _channel.invokeMethod('getTemporaryDirectory');
+    if (path == null)
```

Comment:
You seem to make this test also in the iOS code? Doing it once should be enough. Didn't check Java side.